### ISSUE TITLE
electrumnodeclient

### ISSUE THREAD

**Issue by starrymoonlight** [vault713/grinswap#2]
What is needed on the server side for this code? Are you using electrumx? I see the electrum_server project is discontinued and electrumx seems somewhat unreliable for me.

**Comment by jaspervdm** [comment]
Yes, I've been using ElectrumX servers. Technically we could support other type of BTC APIs by implemeting the [BtcNodeClient trait](https://github.com/vault713/grinswap/blob/master/src/swap/bitcoin/client.rs#L39) for them, but I haven't found any that fit our requirements.

**Comment by starrymoonlight** [comment]
Were you able to get it work on testnet? I got an assertion failure at about block 200,000 every time and electrumx wasn't able to start.
